2/src/part2.c

```diff
@@ -0,0 +1,74 @@
+// Link: https://adventofcode.com/2022/day/2
+
+#include <stdio.h>
+
+enum {
+    ROCK = 1,
+    PAPER,
+    SCISSORS
+};
+
+int decode_move(char move) {
+    switch (move) {
+        case 'A':  // Rock
+            return ROCK;
+        case 'B':  // Paper
+            return PAPER;
+        case 'C':  // Scissors
+            return SCISSORS;
+        default:
+            return 0;
+    }
+}
+
+int choose_your_move(int opponent_move, char desired_outcome) {
+    switch (desired_outcome) {
+        case 'X':  // You need to lose
+            if(opponent_move - 1 > 0){
+                return opponent_move - 1;
+            }else{
+                return SCISSORS;
+            }
+        case 'Y':  // You need a draw
+            return opponent_move;
+        case 'Z':  // You need to win
+            if(opponent_move + 1 < 4){
+                return opponent_move + 1;
+            }else{
+                return ROCK;
+            }        
+        default:
+            return 0;
+    }
+}
+
+int get_score_for_outcome(char outcome) {
```

Comment:
```suggestion
int calculate_round_score(MoveType opponent_move, MoveType your_move) {
```

calculated round score based on your and opponent move
like:
....
switch (opponent_move) {
        case ROCK:
            return (your_move == PAPER) ? 6 : 0;
            .....